cmake/UsePythonTest.cmake

```
@@ -20,96 +19,59 @@
 #
 
 
-MACRO(ADD_PYTHON_TEST TESTNAME FILENAME)
-  GET_SOURCE_FILE_PROPERTY(loc ${FILENAME} LOCATION)
-  GET_SOURCE_FILE_PROPERTY(pyenv ${FILENAME} PYTHONPATH)
+macro(ADD_PYTHON_TEST TESTNAME FILENAME)
+  get_source_file_property(QGIS_PYTEST_FILE_LOC ${FILENAME} LOCATION)
+  get_source_file_property(QGIS_PYTEST_PYENV ${FILENAME} PYTHONPATH)
+
   #Avoid "NOTFOUND" string when setting LD_LIBRARY_PATH later
-  if(EXISTS "${pyenv}")
-    set(pyenv "${pyenv}:")
+  if(EXISTS "${QGIS_PYTEST_PYENV}")
+    set(QGIS_PYTEST_PYENV "${QGIS_PYTEST_PYENV}:")
   else()
-    set(pyenv "")
+    set(QGIS_PYTEST_PYENV "")
   endif()
-  IF(WIN32)
-    STRING(REGEX REPLACE ":" " " wo_semicolon "${ARGN}")
-    IF(USING_NINJA OR USING_NMAKE)
-      FILE(WRITE ${CMAKE_CURRENT_BINARY_DIR}/${TESTNAME}.cmake "
-SET(ENV{QGIS_PREFIX_PATH} \"${QGIS_OUTPUT_DIRECTORY}/bin\")
-SET(ENV{PATH} \"${QGIS_OUTPUT_DIRECTORY}/bin;\$ENV{PATH}\")
-SET(ENV{PYTHONPATH} \"${PYTHON_OUTPUT_DIRECTORY};${PYTHON_OUTPUT_DIRECTORY}/plugins;${CMAKE_SOURCE_DIR}/tests/src/python;\$ENV{PYTHONPATH}\")
-MESSAGE(\"PATH:\$ENV{PATH}\")
-")
-    ELSE(USING_NINJA OR USING_NMAKE)
-      FILE(WRITE ${CMAKE_CURRENT_BINARY_DIR}/${TESTNAME}.cmake "
-SET(ENV{QGIS_PREFIX_PATH} \"${QGIS_OUTPUT_DIRECTORY}/bin/\${CMAKE_BUILD_TYPE}\")
-SET(ENV{PATH} \"${QGIS_OUTPUT_DIRECTORY}/bin/\${CMAKE_BUILD_TYPE};\$ENV{PATH}\")
-SET(ENV{PYTHONPATH} \"${PYTHON_OUTPUT_DIRECTORY};${PYTHON_OUTPUT_DIRECTORY}/plugins;${CMAKE_SOURCE_DIR}/tests/src/python;\$ENV{PYTHONPATH}\")
-MESSAGE(\"PATH:\$ENV{PATH}\")
-")
-    ENDIF(USING_NINJA OR USING_NMAKE)
-  ELSE(WIN32)
-    STRING(REGEX REPLACE ";" " " wo_semicolon "${ARGN}")
-    FILE(WRITE ${CMAKE_CURRENT_BINARY_DIR}/${TESTNAME}.cmake "
-SET(ENV{QGIS_PREFIX_PATH} \"${QGIS_OUTPUT_DIRECTORY}\")
-SET(ENV{LD_LIBRARY_PATH} \"${pyenv}${QGIS_OUTPUT_DIRECTORY}/lib:\$ENV{LD_LIBRARY_PATH}\")
-SET(ENV{PYTHONPATH} \"${PYTHON_OUTPUT_DIRECTORY}:${PYTHON_OUTPUT_DIRECTORY}/plugins:${CMAKE_SOURCE_DIR}/tests/src/python:\$ENV{PYTHONPATH}\")
-MESSAGE(\"export LD_LIBRARY_PATH=\$ENV{LD_LIBRARY_PATH}\")
-")
-  ENDIF(WIN32)
 
-  SET(TEST_TIMEOUT 0)
+  set(_QGIS_PYTEST_PYTHONPATHS "${PYTHON_OUTPUT_DIRECTORY}" "${PYTHON_OUTPUT_DIRECTORY}/plugins" "${CMAKE_SOURCE_DIR}/tests/src/python")
 
-  FOREACH(_in ${ARGN})
-    STRING(REGEX MATCH "^([^=]+)=(.*)$" _out ${_in})
+  if(WIN32)
+    set(QGIS_PYTEST_SEPARATOR ";")
+    set(PATH_VAR_NAME "PATH")
+    if(USING_NINJA OR USING_NMAKE)
```

```suggestion
    if(NOT CMAKE_CONFIGURATION_TYPES)
```

This should be more appropriate?